File: backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("products")]
    public class Product
    {
        [Key] public int Id { get; set; }

        [Required, StringLength(100)]
        public string Name { get; set; }

        public int? CategoryId { get; set; }

        [ForeignKey("CategoryId")]
        public Category? Category { get; set; }

        [Required]
        public decimal Price { get; set; }

        public int Stock { get; set; } = 0;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: backend/src/main/java/pika/hungt1/dx/entity/Product.java
package pika.hungt1.dx.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "products")
@Data @NoArgsConstructor @AllArgsConstructor
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable=false, length=100)
    private String name;

    @ManyToOne
    @JoinColumn(name="category_id")
    private Category category;

    @Column(nullable=false)
    private BigDecimal price;

    @Column(nullable=false)
    private Integer stock = 0;

    @Column(name="created_at")
    private Instant createdAt = Instant.now();
}
